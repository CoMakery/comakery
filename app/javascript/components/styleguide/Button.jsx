import React from "react"
import PropTypes from "prop-types"
import classNames from "classnames"

class Button extends React.Component {
  render () {
    const {className, type, value, disabled, ...other} = this.props

    const classnames = classNames(
      'button',
      className
    )

    return (
      <React.Fragment>
        <input className={classnames} {...other}
          type={type}
          value={value}
          disabled={disabled}
        />
      </React.Fragment>
    )
  }
}

Button.propTypes = {
  className: PropTypes.string,
  type: PropTypes.string,
  value: PropTypes.string,
  disabled: PropTypes.bool
}
Button.defaultProps = {
  className: '',
  type: 'button',
  value: 'publish',
  disabled: false
}
export default Button
