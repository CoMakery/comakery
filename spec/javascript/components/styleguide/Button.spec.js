import React from 'react'
import { shallow, mount, render } from 'enzyme'
import Button from 'components/styleguide/Button'

describe('Button', () => {
  it('renders correctly without props', () => {
    const wrapper = shallow(<Button />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.exists('.button')).toBe(true)
    expect(wrapper.find('.button').props().value).toBe('publish')
    expect(wrapper.find('.button').props().type).toBe('button')
    expect(wrapper.find('.button').props().disabled).toBe(false)
  })

  it('renders correctly with custom className', () => {
    const wrapper = shallow(<Button className="__test" />)

    expect(wrapper.exists('.button.__test')).toBe(true)
  })

  it('renders correctly with custom type', () => {
    const wrapper = shallow(<Button type="file" />)

    expect(wrapper.find('.button').props().type).toBe('file')
  })

  it('renders correctly with custom value', () => {
    const wrapper = shallow(<Button value="save" />)

    expect(wrapper.find('.button').props().value).toBe('save')
  })

  it('renders correctly when disabled', () => {
    const wrapper = shallow(<Button disabled />)

    expect(wrapper.find('.button').props().disabled).toBe(true)
  })
})
