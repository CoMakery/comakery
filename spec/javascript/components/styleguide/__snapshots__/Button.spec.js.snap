// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly without props 1`] = `
<Fragment>
  <input
    className="button"
    disabled={false}
    type="button"
    value="publish"
  />
</Fragment>
`;
